### DISCUSSION TITLE
File not found  The site configured at this address does not contain the requested file.

### DISCUSSION THREAD

**Discussion by temavrn** [community/community#22640] [Other Feature Feedback, Questions, & Ideas]
<p>I follow exactly the tutorial of deploying the site on git-pages, and even have gh-pages branch, but error happens again and again. although i have in my Git Pages settings:</p>
<p>Your site is ready to be published at <a href="https://temavrn.github.io/my_site/" rel="nofollow noopener">https://temavrn.github.io/my_site/</a>.</p>
<p>and i tried hexo-deployer git, but nothing works :frowning:</p>
<p>should i try to install git desktop?</p>
<p>this is my repo <a href="https://github.com/temavrn/my_site" rel="nofollow noopener">https://github.com/temavrn/my_site</a></p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/temavrn">@temavrn</a>,</p>
<p>Your site seems to work for me. Are you still having problems with it? Sometimes it takes a little while to deploy for the first time.</p>
<p>Thanks!</p>

**Comment by temavrn** [comment]
<p>thanks, yes i just started from creating new repository and followed all the instructions. nevertheless, i still don’t get whether it is necessary to have <a href="http://username.github.io" rel="nofollow noopener">username.github.io</a> name of the repository or not</p>

**Comment by temavrn** [comment]
<p>and how can i update the content of the site? I commited, pushed and it is shown on github, but i dont have any changes on my website</p>

**Comment by ThirdEyeRose** [comment, marked as answer]
<p>Hi <a class="mention" href="/temavrn">@temavrn</a>,</p>
<p>I’m not exactly sure what you mean regarding the <a href="http://username.github.io" rel="nofollow noopener">username.github.io</a> requirement. Are you asking about custom domains? If so, you can follow these <a href="https://help.github.com/articles/using-a-custom-domain-with-github-pages/" rel="nofollow noopener">instructions to set up a custom domain for your GitHub Pages site</a>.</p>
<p>As for updating your site, the content should update whenever you make a change and push it up to your GitHub Pages branch. However, this can take a few minutes to process your changes and re-render the site.  If your changes still aren’t showing up, I recommend checking out these <a href="https://help.github.com/articles/troubleshooting-github-pages-builds/" rel="nofollow noopener">steps for troubleshooting GitHub Pages builds</a> and see if that helps or anything.</p>
<p>Thanks!</p>

**Comment by ThirdEyeRose** [comment]
<p>Hi <a class="mention" href="/temavrn">@temavrn</a>,</p>
<p>Would you be able to give an example of what you’re not seeing that you should be seeing as changed?</p>
<p>Thanks!</p>

**Comment by temavrn** [comment]
<p>sorry cant understand what you are asking :slight_smile:</p>
<p>i figured the problem by numerous trials and errors and at the end i just type NPM RUN DEPLOY every time after committing to the gh-pages branch and it works. It took me 3 weeks to figure it out. i still have some crap errors if i try to push to the server like UPSTREAM BRANCH blabla bla you **bleep**ty programmer, but NPM RUN DEPLOY works perfect even if having errors in gitbash while pushing. thanks again anyway</p>

**Comment by ThirdEyeRose** [comment]
<p><a class="mention" href="/temavrn">@temavrn</a>, I’m glad that you figured out something to work even if it’s not a perfect solution. I didn’t know that you were using a Node site with GitHub Pages which is part of why I think I got confused. GitHub Pages is really designed to be used with a static site and is geared towards Jekyll or raw HTML/CSS. This could be a part of the problem in the workflow.</p>

**Comment by temavrn** [comment]
<p>thanks. but i followed some videotutorial of a professional web developer and he used that, so i just followed sometimes without understanding. anyway he got and i got it in the end.</p>
